### PR TITLE
Sorted collections

### PR DESCRIPTION
**Pull request by vihanb**
This introduces two new sorted collections within a `SortedCollections` subpackage as initially 
discussed in #1.

These sorted collections maintain their order by internally maintaining in-memory B-trees which allows them to balance the performance benefits of contiguous buffers and trees.

Currently, this contains an implementation of `SortedSet` and `SortedDictionary` which are designed to mimic the API of the standard library’s `Set` and `Dictionary` as closely as possible and provide additional methods and properties specific to sorted collection types.

### Checklist
- [x] I've read the [Contribution Guidelines](/README.md#contributing-to-swift-collections)
- [x] My contributions are licensed under the [Swift license](/LICENSE.txt).
- [x] I've followed the coding style of the rest of the project.
- [ ] I've added tests covering all new code paths my change adds to the project (if appropriate).
- [ ] I've added benchmarks covering new functionality (if appropriate).
- [x] I've verified that my change does not break any existing tests or introduce unexplained benchmark regressions.
- [ ] I've updated the documentation if necessary.

### CODE REVIEW

File: Sources/SortedCollections/BTree/_BTree.PackedList.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension _BTree {
+  /// A stack-allocated list of some values.
+  /// 
+  /// - Warning: This may hold strong references to objects after they
```
Comment (ishantheperson) [inline Sources/SortedCollections/BTree/_BTree.PackedList.swift:15]: After they what?

File: Sources/SortedCollections/BTree/_BTree+Partial RangeReplaceableCollection.swift [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    }
+    return builder.finish()
+  }
+
+  
+  // MARK: Last Removal
+  
+  /// Removes the first element of a tree, if it exists.
+  ///
+  /// - Returns: The moved first element of the tree.
```
Comment (Dante-Broggi) [inline Sources/SortedCollections/BTree/_BTree+Partial RangeReplaceableCollection.swift:35]:
```suggestion
  /// - Returns: The moved last element of the tree.
```

**Comment by vihanb** [comment]
Our initial benchmark numbers are promising, in most cases we come close to, when not exceeding, the performance at large collection sizes compared to `std::map`

lookups-13
insertions-1

**Comment by lorentey** [comment]
Merging to feature branch `feature/SortedCollections`.

File: Sources/SortedCollections/SortedSet/SortedSet.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+/// A collection which maintains unique members in ascending sorted order.
+public struct SortedSet<Element: Comparable> {
+  @usableFromInline
+  internal typealias _Tree = _BTree<Element, ()>
+  
+  @usableFromInline
+  internal var _root: _Tree
+  
+  //// Creates an empty set.
```
Comment (mdznr) [inline Sources/SortedCollections/SortedSet/SortedSet.swift:20]:
```suggestion
  /// Creates an empty set.
```

**Pull request merged**